api/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,109 @@
+<?php namespace App\Http\Controllers;
+
+use RuntimeException;
+use App\Models\AuthToken;
+use Tymon\JWTAuth\JWTAuth;
+use Illuminate\Http\Request;
+use App\Exceptions\UnauthorizedException;
+use Tymon\JWTAuth\Exceptions\JWTException;
+use Tymon\JWTAuth\Exceptions\TokenExpiredException;
+
+class AuthController extends BaseController
+{
+    /**
+     * JWT Auth
+     *
+     * @var Tymon\JWTAuth\JWTAuth
+     */
+    protected $jwtAuth;
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  JWTAuth  $jwtAuth
+     * @return void
+     */
+    public function __construct(JWTAuth $jwtAuth)
+    {
+        $this->jwtAuth = $jwtAuth;
+    }
+
+    /**
+     * Get a login token.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function login(Request $request)
+    {
+        $credentials = [
+            'email' => $request->getUser(),
+            'password' => $request->getPassword()
+        ];
+
+        try {
+            if (!$token = $this->jwtAuth->attempt($credentials)) {
+
+                throw new UnauthorizedException;
+            }
+        } catch (JWTException $e) {
+            throw new RuntimeException('Token could not be encoded.');
```

Comment:
This happens for instance if an algorithm is configured that the package can not handle. For instance this exception was thrown when we switched to 'RS256' algo, as the package could not support it. As I've now extended the package to support that algo, this exception should theoretically not be thrown anymore. Though, if switching in the config file to another unsupported algo, it would be thrown.